### ISSUE TITLE
Add support for DynamoDB as a peer discovery key value store

### ISSUE THREAD

**Issue by wolfeidau** [honeycombio/refinery#586]
I am interested to hear if others thing there would be value in adding DynamoDB peer discovery support, this would reduce the complexity / infrastructure required when deploying in AWS.

The advantages of supporting DynamoDB are:

* Managed service
* Billed per request so good for those starting out
* Simplify deployment infrastructure for AWS customers

I have implemented DynamoDB in a bunch of similar situations for peer discovery so I am happy to raise a PR.

Cheers

**Comment by kentquirk** [comment]
Hi, @wolfeidau -- this is an interesting idea.

Right now, Refinery doesn't scale dynamically very well under changes in load. Part of the problem is peer discovery and management. We're in the middle of doing a lot of work to improve that, and I'd prefer it if we don't complicate things by adding a new peer management store along the way.

Can we table this for a few weeks while we work out the scalability stuff, please? I do think there's value in what you propose, I'd just prefer to sequence it to minimize the complexity.

**Comment by wolfeidau** [comment]
@kentquirk yeah for sure, thanks for the note with some of the inflight work, keen to see how i can make this easier to deploy and operate as well!

**Comment by alexrudd** [comment]
Hi @kentquirk, just curious if this has been considered since?

The Peers interface looks nice and simple, and it would be a great if there were some serverless backend options (DynamoDB, s3, etc)
